### ISSUE TITLE
Using matplotlib instead of gnuplot?

### ISSUE THREAD

**Issue by JanaKoehler1985** [mzechmeister/viper#36]
A question that came up lately by a few users, why do we use gnuplot instead of matplotlib. And would it be possible to change everything to matplotlib.

I think we also had this discussion once while thinking about showing plots using a GUI. Would it be worth it to think about switching to matplotlib? Would it easily doable?

I start to like gnuplot and definitely see some benefits in it, on the other hand, matplotlib seems to be more common to the community.

**Comment by mzechmeister** [comment]
I was afraid of such a request.

matplotlib has become quasi the default plot library in python and many like it. Yet, I never fell in love with matlibplot. Syntax to hard to remember.
Well, we now about the pros and cons.

BTW, gnuplot could be also embedded into a tk gui, see here
https://github.com/mzechmeister/xgp ;)
But not qt or wxt terminal; so it won't be shiny or speedy.

I also would like to mention another alternative worth to consider: plotly.
Syntax might be even more complicated than matplotlib(!)
Plots will appear in the browser and it could coupled be with html guis (instead of tkinter).
Here are two plotly project that I made directly with javascript
https://github.com/mzechmeister/csvplotter
www.astro.physik.uni-goettingen.de/research/solar-lib
So, plotting a lot of spectra with nice interactivity is possible.
There is also the wrapper plotly-python and plotly-dash.

Back to matplotlib; what else to know:
* I know of two serval forks that have replaced gnuplot by matplotlib.
* I have seen gnuplot in some ESO projects.

In summary, I would not block to go with matplotlib, but I don't have the ambition to spent my time on the conversion.

**Comment by JanaKoehler1985** [comment]
I knew you wouldn't be happy about this request. I am not happy about it either and was at least able to push it down on the priority list of the people asking for it. But I promised to ask you ;-)

I will have a look at your gnuplot-gui option. This sounds interesting.

I also would prefer not to change from gnuplot to matplotlib. Not because I don't like it, but because I am also not interested in changing it, nor I see (at the moment) any benefits coming along with it. 

> I have seen gnuplot in some ESO projects.

Do you, by an chance, remember the name of these projects? This would be good to know.

**Comment by mzechmeister** [comment]
FYI, I have created now [csv_sent.py](https://gist.github.com/mzechmeister/8bfec277c385c6d72fbc9c8dcedbb2fd).
It can be also use inspect the output with csv_plotter, e.g. `csv_sent tmp.rvo.dat` or  `csv_sent tmp.par.dat`.
